homeassistant/components/updater/__init__.py

```diff
@@ -99,18 +111,17 @@ def _load_uuid(hass, filename=UPDATER_UUID_FILE):
             newest = hass.components.hassio.get_homeassistant_version()
 
         # Validate version
+        update_available = False
         if StrictVersion(newest) > StrictVersion(current_version):
-            _LOGGER.info("The latest available version is %s", newest)
-            hass.states.async_set(
-                ENTITY_ID,
-                newest,
-                {
-                    ATTR_FRIENDLY_NAME: "Update Available",
-                    ATTR_RELEASE_NOTES: releasenotes,
-                },
-            )
+            _LOGGER.info("The latest available version of Home Assistant is %s", newest)
+            update_available = True
         elif StrictVersion(newest) == StrictVersion(current_version):
             _LOGGER.info("You are on the latest version (%s) of Home Assistant", newest)
+        elif StrictVersion(newest) < StrictVersion(current_version):
+            _LOGGER.debug("Local version is newer than the latest version (%s)", newest)
+
+        updater = Updater(update_available, newest, release_notes)
+        async_dispatcher_send(hass, DISPATCHER_REMOTE_UPDATE, updater)
```

Comment:
Before we send the first update we need to load the binary sensor platform via `discovery.async_load_platform`. Schedule that call with `hass.async_create_task`.

Reply:
Unfortunately I couldn't find a description of the required parameters. I have filled them in as good as I could, but I missed the fourth one.

Reply:
Looks good!

Reply:
```
2019-08-02 00:04:45 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/home/tobi/workspace/home-assistant/homeassistant/helpers/discovery.py", line 143, in async_load_platform
    assert hass_config, "You need to pass in the real hass config"
AssertionError: You need to pass in the real hass config
```

I'm afraid there is still a mistake

Reply:
Found it!